streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -238,14 +237,14 @@ class FileBasedWriteAheadLogSuite
     val executionContext = ExecutionContext.fromExecutorService(fpool)
 
     class GetMaxCounter {
-      private val value = new AtomicInteger()
+      private var value = 0
       @volatile private var max: Int = 0
       def increment(): Unit = synchronized {
-        val atInstant = value.incrementAndGet()
+        val atInstant = value + 1
         if (atInstant > max) max = atInstant
       }
-      def decrement(): Unit = synchronized { value.decrementAndGet() }
-      def get(): Int = synchronized { value.get() }
+      def decrement(): Unit = synchronized { value = value - 1 }
+      def get(): Int = synchronized { value }
       def getMax(): Int = synchronized { max }
```

Comment:
Actually, this suite failed in the CI with your PR, @beliefer .
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 440, Failed 1, Errors 0, Passed 439, Ignored 1
[error] Failed tests:
[error] 	org.apache.spark.streaming.util.FileBasedWriteAheadLogSuite
```

Reply:
Given the failure, I'm not sure if this test suite change is worthy or not.

Reply:
Thank you for your reminder. I will check it.

Reply:
I can't see why this change would be different. But there is probably some subtle issue with memory barriers or whatever. I don't think it's worth changing.

Reply:
No. I have investigated. The reason is the earlier change is not correct. It's not the issue with memory barriers.
The earlier change is from
```
val atInstant = value.incrementAndGet()
if (atInstant > max) max = atInstant
```
to
```
val atInstant = value + 1
if (atInstant > max) max = atInstant
```

Reply:
Now I fixed it.